compiler/rustc_interface/src/passes.rs

```diff
@@ -222,7 +223,8 @@ fn configure_and_expand(mut krate: ast::Crate, resolver: &mut Resolver<'_, '_>)
 
         // Create the config for macro expansion
         let features = sess.features_untracked();
-        let recursion_limit = get_recursion_limit(&krate.attrs, sess);
+        let attrs = krate.attrs.iter().map(|at| (at.name_or_empty(), at)).collect();
+        let recursion_limit = get_recursion_limit(&attrs, sess);
```

Comment:
This should perhaps be left as-is, to avoid constructing a data structure for just one use.